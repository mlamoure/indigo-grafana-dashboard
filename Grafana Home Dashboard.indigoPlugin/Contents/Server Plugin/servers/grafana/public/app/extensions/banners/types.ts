import React from 'react';

import { BusEventWithPayload } from '@grafana/data';

/** @alpha */
export class PageBannerDisplayEvent extends BusEventWithPayload<PageBannerEventPayload> {
  static type = 'page-banner-display';
}

/** @alpha */
export type PageBannerEventPayload = {
  onClose?: () => void;
  severity?: PageBannerSeverity;
  body: React.ElementType<PageBannerBodyProps>;
};

/** @alpha */
export type PageBannerBodyProps = {
  severity: PageBannerSeverity;
};

/** @alpha */
export type PageBannerAction = {
  text: string;
  href: string;
};

/** @alpha */
export enum PageBannerSeverity {
  info,
  warning,
  error,
}
