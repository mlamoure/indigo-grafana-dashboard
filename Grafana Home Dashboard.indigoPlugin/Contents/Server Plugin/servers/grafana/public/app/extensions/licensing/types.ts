export interface LicenseToken {
  status: number;
  jti: string;
  iss: string;
  sub: string;
  iat: number;
  exp: number;
  nbf: number;
  lexp: number;
  lid: string;
  limit_by: string;
  included_admins: number;
  included_viewers: number;
  included_users: number;
  prod: string[];
  company: string;
  usage_billing?: boolean;
  slug: string;
  details_url?: string;
  account?: string;
  trial_exp?: number;
}

export interface ActiveUserStats {
  active_admins_and_editors: number;
  active_viewers: number;
  active_users: number;
}

export type PermissionsReport = {
  id: number;
  title: string;
  slug: string;
  uid: string;
  url: string;
  isFolder: boolean;
  orgId: number;
  granteeType: string;
  granteeName: string;
  granteeUrl: string;
  customPermissions: string;
  orgRole: string;
  usersCount: number;
};
