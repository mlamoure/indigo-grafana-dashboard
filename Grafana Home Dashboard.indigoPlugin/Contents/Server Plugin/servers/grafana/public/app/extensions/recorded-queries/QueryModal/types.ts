import { DataQuery, DataSourceInstanceSettings } from '@grafana/data';

export interface QueryModalModel {
  title: string;
  body: QueryModalBody;
}

export interface QueryModalBodyProps {
  query?: DataQuery;
  queries?: DataQuery[];
  onAddQuery?: (q: DataQuery) => void;
  onChangeDataSource?: (ds: DataSourceInstanceSettings) => void;
}

export type QueryModalBody = React.ComponentType<QueryModalBodyProps>;
