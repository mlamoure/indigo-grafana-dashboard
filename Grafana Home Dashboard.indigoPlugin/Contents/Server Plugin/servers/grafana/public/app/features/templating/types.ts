import { assignModelProperties } from 'app/core/utils/model_utils';
import { Deferred } from '../../core/utils/deferred';
import { VariableModel as BaseVariableModel } from '@grafana/data';

export enum VariableRefresh {
  never,
  onDashboardLoad,
  onTimeRangeChanged,
}

export enum VariableHide {
  dontHide,
  hideLabel,
  hideVariable,
}

export enum VariableSort {
  disabled,
  alphabeticalAsc,
  alphabeticalDesc,
  numericalAsc,
  numericalDesc,
  alphabeticalCaseInsensitiveAsc,
  alphabeticalCaseInsensitiveDesc,
}

export interface VariableTag {
  selected: boolean;
  text: string | string[];
  values?: any[];
  valuesText?: string;
}

export interface VariableOption {
  selected: boolean;
  text: string | string[];
  value: string | string[];
  isNone?: boolean;
  tags?: VariableTag[];
}

export interface AdHocVariableFilter {
  key: string;
  operator: string;
  value: string;
  condition: string;
}

export interface AdHocVariableModel extends VariableModel {
  datasource: string | null;
  filters: AdHocVariableFilter[];
}

export interface IntervalVariableModel extends VariableWithOptions {
  auto: boolean;
  auto_min: string;
  auto_count: number;
  refresh: VariableRefresh;
}

export interface CustomVariableModel extends VariableWithMultiSupport {}

export interface DataSourceVariableModel extends VariableWithMultiSupport {
  regex: string;
  refresh: VariableRefresh;
}

export interface QueryVariableModel extends DataSourceVariableModel {
  datasource: string | null;
  definition: string;
  sort: VariableSort;
  tags: VariableTag[];
  tagsQuery: string;
  tagValuesQuery: string;
  useTags: boolean;
  queryValue?: string;
}

export interface TextBoxVariableModel extends VariableWithOptions {}

export interface ConstantVariableModel extends VariableWithOptions {}

export interface VariableWithMultiSupport extends VariableWithOptions {
  multi: boolean;
  includeAll: boolean;
  allValue?: string | null;
}

export interface VariableWithOptions extends VariableModel {
  current: VariableOption;
  options: VariableOption[];
  query: string;
}

export interface VariableModel extends BaseVariableModel {
  id?: string; // only exists for variables in redux state
  global?: boolean; // only exists for variables in redux state
  hide: VariableHide;
  skipUrlSync: boolean;
  index?: number;
  initLock?: Deferred | null;
}

export interface VariableActions {
  setValue(option: any): any;
  updateOptions(searchFilter?: string): any;
  dependsOn(variable: any): any;
  setValueFromUrl(urlValue: any): any;
  getValueForUrl(): any;
  getSaveModel(): any;
}

export type CtorType = new (...args: any[]) => {};

export interface VariableTypes {
  [key: string]: {
    name: string;
    ctor: CtorType;
    description: string;
    supportsMulti?: boolean;
  };
}

export let variableTypes: VariableTypes = {};
export { assignModelProperties };
