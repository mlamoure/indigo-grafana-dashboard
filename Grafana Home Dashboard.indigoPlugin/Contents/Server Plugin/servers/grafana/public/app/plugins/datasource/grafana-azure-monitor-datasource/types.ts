import { DataQuery } from '@grafana/ui/src/types';

export interface AzureMonitorQuery extends DataQuery {
  format: string;
  subscription: string;
  azureMonitor: AzureMetricQuery;
  azureLogAnalytics: AzureLogsQuery;
  //   appInsights: any;
}

export interface AzureMetricQuery {
  resourceGroup: string;
  resourceName: string;
  metricDefinition: string;
  metricName: string;
  timeGrainUnit: string;
  timeGrain: string;
  timeGrains: string[];
  aggregation: string;
  dimension: string;
  dimensionFilter: string;
  alias: string;
}

export interface AzureLogsQuery {
  query: string;
  resultFormat: string;
  workspace: string;
}

// Azure Log Analytics types
export interface KustoSchema {
  Databases: { [key: string]: KustoDatabase };
  Plugins: any[];
}
export interface KustoDatabase {
  Name: string;
  Tables: { [key: string]: KustoTable };
  Functions: { [key: string]: KustoFunction };
}

export interface KustoTable {
  Name: string;
  OrderedColumns: KustoColumn[];
}

export interface KustoColumn {
  Name: string;
  Type: string;
}

export interface KustoFunction {
  Name: string;
  DocString: string;
  Body: string;
  Folder: string;
  FunctionKind: string;
  InputParameters: any[];
  OutputColumns: any[];
}

export interface AzureLogsVariable {
  text: string;
  value: string;
}

export interface AzureLogsTableData {
  columns: AzureLogsTableColumn[];
  rows: any[];
  type: string;
  refId: string;
  meta: {
    query: string;
  };
}

export interface AzureLogsTableColumn {
  text: string;
  type: string;
}
