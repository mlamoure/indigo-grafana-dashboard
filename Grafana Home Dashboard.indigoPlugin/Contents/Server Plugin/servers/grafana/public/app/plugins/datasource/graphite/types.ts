import { DataQuery } from '@grafana/data';

export interface GraphiteQuery extends DataQuery {
  target?: string;
}

export enum GraphiteType {
  Default = 'default',
  Metrictank = 'metrictank',
}
