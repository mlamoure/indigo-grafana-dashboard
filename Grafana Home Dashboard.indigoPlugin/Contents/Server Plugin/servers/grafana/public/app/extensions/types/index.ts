export * from './permissions';
export * from './store';
export * from './reports';
export * from './metaanalytics';
export * from './caching';
export * from './accessControl';
export * from './recordedQuery';
