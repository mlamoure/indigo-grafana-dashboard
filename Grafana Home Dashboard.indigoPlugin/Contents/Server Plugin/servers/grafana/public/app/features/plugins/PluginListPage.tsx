import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import PageHeader from '../../core/components/PageHeader/PageHeader';
import PluginActionBar from './PluginActionBar';
import PluginList from './PluginList';
import { NavModel, Plugin } from '../../types';
import { loadPlugins } from './state/actions';
import { getNavModel } from '../../core/selectors/navModel';
import { getLayoutMode, getPlugins } from './state/selectors';
import { LayoutMode } from '../../core/components/LayoutSelector/LayoutSelector';

export interface Props {
  navModel: NavModel;
  plugins: Plugin[];
  layoutMode: LayoutMode;
  loadPlugins: typeof loadPlugins;
}

export class PluginListPage extends PureComponent<Props> {
  componentDidMount() {
    this.fetchPlugins();
  }

  async fetchPlugins() {
    await this.props.loadPlugins();
  }

  render() {
    const { navModel, plugins, layoutMode } = this.props;

    return (
      <div>
        <PageHeader model={navModel} />
        <div className="page-container page-body">
          <PluginActionBar />
          {plugins && <PluginList plugins={plugins} layoutMode={layoutMode} />}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    navModel: getNavModel(state.navIndex, 'plugins'),
    plugins: getPlugins(state.plugins),
    layoutMode: getLayoutMode(state.plugins),
  };
}

const mapDispatchToProps = {
  loadPlugins,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(PluginListPage));
