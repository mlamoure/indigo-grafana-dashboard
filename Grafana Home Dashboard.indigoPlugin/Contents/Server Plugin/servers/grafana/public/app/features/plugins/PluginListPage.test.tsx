import React from 'react';
import { shallow } from 'enzyme';
import { PluginListPage, Props } from './PluginListPage';
import { NavModel, Plugin } from '../../types';
import { LayoutModes } from '../../core/components/LayoutSelector/LayoutSelector';

const setup = (propOverrides?: object) => {
  const props: Props = {
    navModel: {} as NavModel,
    plugins: [] as Plugin[],
    layoutMode: LayoutModes.Grid,
    loadPlugins: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<PluginListPage {...props} />);
  const instance = wrapper.instance() as PluginListPage;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
