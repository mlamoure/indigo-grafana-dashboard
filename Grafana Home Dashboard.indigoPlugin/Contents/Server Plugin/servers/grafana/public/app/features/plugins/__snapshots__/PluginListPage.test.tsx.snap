// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <Connect(PluginActionBar) />
    <PluginList
      layoutMode="grid"
      plugins={Array []}
    />
  </div>
</div>
`;
